[WIP]chat-space非同期通信

azunyan0720jp opened the pull request:
#WHAT
chat-spaceの非同期通信を実装。
#WHY
chat-spaceをHTMLでもJSONでも受け取れるようにし、即メッセージが反映されるようにするため。

azunyan0720jp commented:
> .DS_Store のファイルが変更差分に含まれているので、削除しましょう

初歩的で恐縮なのですが、この方法で一括削除して再度プルリクエストした方がよろしいでしょうか？
https://qiita.com/supersnack/items/157b69589e36f340bb29

azunyan0720jp commented:
> .DS_Store のファイルが変更差分に含まれているので、削除しましょう

ただいま削除いたしました！引き続きよろしくお願いいたします！

miyagit commented on app/views/layouts/application.html.haml:
```diff
@@ -0,0 +1,10 @@
+%html
+  %head
+    %meta{:content => "text/html; charset=UTF-8", "http-equiv" => "Content-Type"}/
+    %title ChatSpace
```
ハッシュロケットを直しましょう。

azunyan0720jp replied:
ハッシュロケットを修正しました！

miyagit replied:
コメントされた部分が修正されていないのですがプッシュのし忘れでしょうか？

azunyan0720jp replied:
ハッシュロケットを修正しました！

miyagit replied:
"http-equiv" : "Content-Type"

"http-equiv": "Content-Type"

のほうが良さそうですね。

azunyan0720jp replied:
"http-equiv" : "Content-Type"を"http-equiv": "Content-Type"に書き換えました。

miyagit commented on app/views/layouts/mailer.text.haml:
```diff
@@ -0,0 +1 @@
+= yield
```
mailer系のファイルを削除しましょう

azunyan0720jp replied:
mailer系のファイルを削除しました！

azunyan0720jp replied:
mailer系のファイルを削除しました！

azunyan0720jp committed:
chat-space非同期通信

#WHAT
chat-spaceの非同期通信を実装。
#WHY
chat-spaceをHTMLでもJSONでも受け取れるようにし、即メッセージが反映されるようにするため。

azunyan0720jp commented:
動作画像のキャプチャです。
https://gyazo.com/ee9f5b20ce25a60d2747f4e5fd9dfbf8
https://gyazo.com/cd0c702104440c591f710598189ce675

TOMARUMARU commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,47 @@
+      contentType: false
+    })
+    .done(function(formData) {
+      console.log(formData)
```
console.logは必ず消しましょう！

azunyan0720jp replied:
console.logを削除しました！

TOMARUMARU commented on app/controllers/messages_controller.rb:
```diff
@@ -10,11 +10,11 @@ def index
+        format.html { redirect_to group_messages_path(params[:group_id]) }
+        format.json 
+      end
+      else
```
elseは不要そうです！

azunyan0720jp replied:
elseを削除しました！

TOMARUMARU reviewed (commented):
public/uploads以下はignoreしましょう。
またこちらのPRにも含まれないようにしましょう

azunyan0720jp commented:
> ignore

失礼しました！igonoreしました！

TOMARUMARU commented:
pushをお忘れずにお願いします！

azunyan0720jp commented:
> pushをお忘れずにお願いします！

失礼しました！今プッシュしました！

TOMARUMARU commented:
まだ反映されていないようです...!

azunyan0720jp committed:
else削除、console.log削除

azunyan0720jp commented:
> まだ反映されていないようです...!

失礼しました！今プッシュしました！

TOMARUMARU commented:
> public/uploads以下はignoreしましょう。
またこちらのPRにも含まれないようにしましょう

こちらもお願いします！

azunyan0720jp commented:
> > public/uploads以下はignoreしましょう。
> > またこちらのPRにも含まれないようにしましょう
> 
> こちらもお願いします！

こちらをみてigonoreしているのですが、必要なファイルまで削除してしまいそうです。https://qiita.com/yuuAn/items/b1d1df2e810fd6b92574

TOMARUMARU commented:
最悪コミットを戻すや、手動で削除等でも大丈夫です！

azunyan0720jp committed:
ignored

azunyan0720jp commented:
> 最悪コミットを戻すや、手動で削除等でも大丈夫です！

ありがとうございます！手動で削除しました！

TOMARUMARU commented:
あ、ignoreに追加するのはあくまでpublic uploads以下、という記述で十分で、こちらのPRのファイルを消していただきたいです！
よくわからなければ現場のメンターに見てもらってください！

azunyan0720jp committed:
Update .gitignore

azunyan0720jp committed:
delete_public_files

shingoteshima commented on app/controllers/messages_controller.rb:
```diff
@@ -10,11 +10,10 @@ def index
+      respond_to do |format| 
+        format.html { redirect_to group_messages_path(params[:group_id]) }
+        format.json 
+      end
```
こちらsaveに失敗したときの処理も書いた方が良いですね。

azunyan0720jp replied:
format.html { redirect_to group_messages_path(params[:group_id]) }
        format.json 
      end
      else
こちら、else消した方がいいとのことで、削除したのですが、elseをもう一度つけた方が良いでしょうか？

shingoteshima replied:
elseに処理が書いてなかったのでその場合は不要となりますが、エラーハンドリングとしてsaveに失敗したときの処理をelseに書いておいた方がよいです。

azunyan0720jp committed:
add_else

shingoteshima commented:
elseの中に処理を書きましょう

azunyan0720jp committed:
add_else_action

azunyan0720jp commented:
> elseの中に処理を書きましょう

elseの中に処理をかきました！

TOMARUMARU commented on app/views/messages/create.json.jbuilder:
```diff
@@ -0,0 +1,5 @@
+json.content  @message.content
+json.user_id  @message.id
```
user_idは使用してないと思うので、こちら消しましょう！

azunyan0720jp replied:
json.user_id  @message.idを削除しました！

azunyan0720jp committed:
deleted_user_id

TOMARUMARU reviewed (approved):
LGTM!!
